src/analysisd/decoders/syscollector.c

```diff
@@ -840,8 +846,8 @@ int decode_netinfo(Eventinfo *lf, cJSON * logJSON, int *socket) {
         msg_type = cJSON_GetObjectItem(logJSON, "type")->valuestring;
 
         if (!msg_type) {
-            merror("Invalid message. Type not found.");
-            goto end;
+            merror("Invalid message. Type not found."); // LCOV_EXCL_LINE
+            goto end;                                   // LCOV_EXCL_LINE
```

Comment:
Why are we not testing this and ignoring instead?

Reply:
Is not posible to achieve this section of code, because the only one entry point for  decoder_XXX functions is by call on DecodeSyscollector().

Reply:
I think it is possible but we would need to mock this `json_type = cJSON_GetObjectItem(logJSON, "type");` it would affect the entire file and probably not worth it. The team may decide this.

Reply:
@Dwordcito @pereyra-m

Reply:
Are you sure we can't send a JSON without `type`  to force `cJSON_GetObjectItem()` to return **NULL**?

> it would affect the entire file and probably not worth it.

Wrapping this function directly would indeed generate too many changes. Every test should be updated.

The only thing I can think of is updating `__wrap_cJSON_GetObjectItem()` to allow selecting when we want the real function and when we need the fake one.

Something like
```c
if(test_mode) {
    //fake method
    ...
    return mock;
}
else {
    //real method
    __real_cJSON_GetObjectItem();

}

```

Reply:
Let's not apply changes around this, this is a candidate for refactoring, it doesn't make sense to pass a string for a check that is done from the caller. A boolean would suffice to identify the end message.